### DISCUSSION TITLE
Deserialize field into multiple fields

### DISCUSSION THREAD

**Discussion by codectl** [pydantic/pydantic#4994] [Question]
Suppose I have the following model:

```python
class Person(BaseModel):
    """some dumb class that I came up with to showcase something similar I am trying to do
    """"
    name: str
    is_children: bool
    is_teenager: bool
    is_adult: bool
```

Then if I want to serialize some data:

```python
Person.parse_obj({"name": "Joe", age: 10})
# name=Joe is_children=True is_teenager=False is_adult=False
```

What is the *pydantic* way to map the value of field ```age``` into the object fields?

**Comment by codectl** [comment]
I haven't found any solution yet ....

I am trying to use a "private" variable to do it but no success:

```python
class Person(BaseModel):
    name: str
    is_children: bool
    is_teenager: bool
    is_adult: bool
    _age: int = Field(0, alias="age")

    class Config:
        underscore_attrs_are_private = True

    @root_validator
    def parse_age(cls, values):
        # the value from '_age' is not visible in 'values'
        return values
```

Feels like a workaround but still not getting anywhere ..

**Reply by uriyyo** [reply]
I guess you need to use `pre=True` root_validator:
```py
from pydantic import BaseModel, Field, root_validator


class Person(BaseModel):
    name: str

    is_children: bool
    is_teenager: bool
    is_adult: bool

    _age: int = Field(0, alias="age")

    class Config:
        underscore_attrs_are_private = True

    @root_validator(pre=True)
    def parse_age(cls, values):
        match values:
            case {"age": age}:
                values.update(
                    is_children=age < 18,
                    is_teenager=18 <= age < 25,
                    is_adult=age >= 25,
                )

        return values


p = Person(name="John", age=20)
print(p)
```
```py
name='John' is_children=False is_teenager=True is_adult=False
```

**Reply by codectl** [reply]
Hi @uriyyo , thank you for your answer ! 

In your approach, having the field ```_age``` is redundant, being there or not makes no difference. That said, any way to accomplish this but with no ```pre=True``` enabled? That is what I was trying to do with a private attribute for deserialization.

I ended up doing something like this:

```python
class Person(BaseModel):
    name: str
    person_id: int = Field(0, alias="personId")
    driver_license_id: int = Field(0, alias="driverLicenseId")
    ...

    @root_validator(pre=False)
    def parse_events(cls, values):
        person_id = values.pop("person_id", None)
        driver_license_id = values.pop("driver_license_id", None)
        # do something with person id and driver license id, etc ...
        return values
```

As you see, I prefer the ```pre=False``` because I don't want to use the ```alias``` field to do a match case over raw fields. Sometime the data to be deserialized can have weird values and the ```root_validator``` is not a good place to perform that check.

The approach I presented works but it requires me to define a "temporary" field to hold the field data that I then ```pop```. For that reason I was looking for a private field but ```values``` do not include that information.
